test.py

```diff
@@ -0,0 +1,172 @@
+import json
+from tqdm import tqdm
+
+list = []
+def fetch_article_details(article_url):
+    article_response = requests.get(article_url)
+    dictionary = {}
+    if article_response.status_code == 200:
+        article_soup = BeautifulSoup(article_response.content, 'html.parser')
+        h1_tag = article_soup.find('h1', class_='arttitle')
+        h2_tag = article_soup.find('h2', class_='artdec')
+        img_tag = article_soup.find('img', class_='lazy')
+        dictionary["URL"] = article_url
+        if h1_tag:
+            dictionary["Title"] = h1_tag.text.strip()
+            if h2_tag: 
+                dictionary["Summary"] = h2_tag.text.strip()     
+                if img_tag:
+                    img_src = img_tag['src']
+                    if img_src == "https://static.indiatvnews.com/ins-web/images/lazy-big.jpg":
+                        dictionary.update({"Image URL": "Unable to fetch"})
+                        dictionary["Image URL"] = "Unabel to fetch"
+                        pass
+                    else:
+                        dictionary.update({"Image URL": img_src})
+                        dictionary["Image URL"] = img_src
+                        pass
+                else:
+                    print("No images found")
+    else:
+        print(f'Failed to retrieve the article. Status code: {article_response.status_code}')
+        print("Article URL:", article_url)
+    list.append(dictionary)
+    
+
+def scrape_and_fetch(link):
+    response = requests.get(link)
+
+    if response.status_code == 200:
+        soup = BeautifulSoup(response.content, 'html.parser')
+        div_tag = soup.find('div', class_='row lhsBox sport_tnews mb20')
+
+        if div_tag:
+            thumb_links = div_tag.find_all('a', class_='thumb')
+
+            for thumb_link in thumb_links:
+                article_url = thumb_link.get('href')
+                fetch_article_details(article_url)
+        else:
+            print('Failed to find the div with class="row lhsBox sport_tnews mb20".')
+    else:
+        print(f'Failed to retrieve the webpage. Status code: {response.status_code}')
+        print("scrape_and_fetch", link)
+
+
+def scrape_and_fetch2(link):
+    response = requests.get(link)
+
+    if response.status_code == 200:
+        soup = BeautifulSoup(response.content, 'html.parser')
+        div_tag = soup.find('div', class_='row lhsBox s_two_column pt20')
+
+        if div_tag:
+            thumb_links = div_tag.find_all('a', class_='thumb')
+
+            for thumb_link in thumb_links:
+                article_url = thumb_link.get('href')
+                fetch_article_details(article_url)
+        else:
+            print('Failed to find the div with class="row lhsBox s_two_column pt20".')
+    else:
+        print(f'Failed to retrieve the webpage. Status code: {response.status_code}')
+        print("scrape_and_fetch2", link)
+
+
+def scrape_and_fetch3(link):
+    response = requests.get(link)
```

Comment:
Create a new function `fetch_page` that is responsible for getting a page rather than duplicating the code.